pkg/config/config.go

```diff
@@ -16,16 +18,19 @@ import (
 )
 
 // Values returns the Value struct from the linkerd-config ConfigMap
-func Values(filepath string) (*l5dcharts.Values, error) {
-	values := &l5dcharts.Values{}
-	configYaml, err := ioutil.ReadFile(filepath)
+func Values(path string, prefix string) (*l5dcharts.Values, error) {
+	if !strings.HasPrefix(path, prefix) {
```

Comment:
What is prefix for?  It appears to be unused except in this validation?

Reply:
That's correct its purpose is validation. The one place we use it is validating the proxy-injector [valuesConfig](https://github.com/linkerd/linkerd2/blob/1fcfe39e9000ac3de2790261bf7d7b1faa366902/controller/proxy-injector/webhook.go#L43). This validation is in addition to the `filepath.Clean(path)` (which is the only one necessary to satisfy `gosec`).

Reply:
I don't really understand why we're doing anything at all with `pkgK8s.MountPathBase`.  This constant isn't directly used anywhere, it's simply used to build up the mount path constants.  Why does it need to be validated?  (and why here?)

Reply:
It doesn't add any extra validation I removed it.